### ISSUE TITLE
PHP 8.2 Creation of dynamic property

### ISSUE THREAD

**Issue by ipajen** [wpengine/phpcompat#318]
### Environment
- WordPress: 6.2
- PHP: 8.2.3
- Server: LiteSpeed
- Database: mysqli (Server: 10.4.28-MariaDB / Client: mysqlnd 8.2.3)
- Browser: Edge 111.0.1661.62 (Windows 10/11)
- Theme: GeneratePress 3.3.0
- MU-Plugins:
  * Health Check Troubleshooting Mode 1.9.0
  * WordPress automation by Installatron
- Plugins:
  * PHP Compatibility Checker 1.6.1
  * Plugin Compatibility Checker 1.9
  * Query Monitor 3.12.1
  * WordPress Beta Tester 3.3.7

```
Creation of dynamic property WPEngine_PHPCompat\PHP_Compatibility_Checker::$page is deprecated
wp-content/plugins/php-compatibility-checker/lib/class-php-compatibility-checker.php:284
```

**Comment by mindctrl** [comment]
@ipajen thank you for the report.

@anthonyburchell here's a good argument for adding 8.1 and 8.2 support ASAP.

**Comment by ipajen** [comment]
@mindctrl  Yes please fixing the warning in this plugin is a great start. Thank you.

**Comment by EmSixTeen** [comment]
Installed this plugin in the hope of sorting this in another plugin in Query Monitor, only to get this same deprecation error. :)
